packages/gatsby/cache-dir/fast-refresh-overlay/components/overlay.js

```diff
@@ -1,43 +1,142 @@
-import React from "react"
+import * as React from "react"
+import { lock, unlock } from "../helpers/lock-body"
+import a11yTrap from "../helpers/focus-trap"
 
-export default function Overlay({ header, body, dismiss }) {
+function Backdrop() {
+  return <div data-gatsby-overlay="backdrop" />
+}
+
+export function VisuallyHidden({ children }) {
+  return (
+    <span
+      style={{
+        border: 0,
+        clip: `rect(0 0 0 0)`,
+        height: `1px`,
+        margin: `-1px`,
+        overflow: `hidden`,
+        padding: 0,
+        position: `absolute`,
+        width: `1px`,
+        whiteSpace: `nowrap`,
+        wordWrap: `normal`,
+      }}
+    >
+      {children}
+    </span>
+  )
+}
+
+export function Overlay({ children }) {
+  React.useEffect(() => {
+    lock()
+
+    return () => {
+      unlock()
+    }
+  }, [])
+
+  const [overlay, setOverlay] = React.useState(null)
+  const onOverlay = React.useCallback(el => {
+    setOverlay(el)
+  }, [])
+
+  React.useEffect(() => {
+    if (overlay === null) {
+      return
+    }
+
+    const handle = a11yTrap({ context: overlay })
+
+    // eslint-disable-next-line consistent-return
+    return () => {
+      handle.disengage()
+    }
+  }, [overlay])
+
+  return (
+    <div data-gatsby-overlay="wrapper" ref={onOverlay}>
+      <Backdrop />
+      <div
+        data-gatsby-overlay="root"
+        role="alertdialog"
```

Comment:
Purposefully chose alertdialog here and not dialog.

> The difference with regular dialogs is that the alertdialog role should only used when an alert, error, or warning occurs. In other words, when a dialog's information and controls require the user's immediate attention alertdialog should be used instead of dialog. It is up to the developer to make this distinction though.

This fits our use case here